src/components/LoadingIndicator.tsx

```diff
@@ -1,13 +1,31 @@
+import { useFetcher } from "@remix-run/react"
 import clsx from "clsx"
+import { useEffect, useMemo } from "react"
+import type { AnalyzationStatus } from "~/analyzer/analyze.server"
 import anitruck from "~/assets/truck.gif"
 
-export function LoadingIndicator({
-  loadingText = "Loading",
-  className = ""
-}: {
-  loadingText?: string
-  className?: string
-}) {
+interface ProgressData {
```

Comment:
Prefer type over interfaces

https://www.totaltypescript.com/type-vs-interface-which-should-you-use